File: Backend/WebShopAPI/Models/Data/Product.cs
namespace WebShopAPI.Models.Data
{
    public record Product
    {
        public int Id { get; init; }
        public required string Name { get; init; }
        public string? Description { get; init; }
        public string? Image {  get; init; }
        public string? Manufacturer { get; init; }
        public decimal Price { get; init; }

        public virtual required ProductCategory Category { get; init; }
    }
}

File: Frontend/src/objects/Product.ts
import { ProductCategory } from "./ProductCategory";

export class Product {
    constructor(
        public readonly id: number,
        public readonly name: string,
        public readonly description: string,
        public readonly manufacturer: string,
        public readonly image: string,
        public readonly price: number,
        public readonly category: ProductCategory
    )
    {}
}

export default Product